test/hotspot/gtest/x86/x86-asmtest.py

```diff
@@ -624,6 +624,14 @@ def generate(RegOp, ops, print_lp64_flag=True, full_set=False):
                 instr = RegOp(*op, reg1=test_reg1, reg2=test_reg2, imm=imm)
                 print_instruction(instr, lp64_flag, print_lp64_flag)
 
+                # additional tests with rax as destination
+                if RegOp in [RegRegImmNddInstruction]:
+                    test_reg1 = 'rax'
+                    test_reg2 = random.choice(test_regs)
```

Comment:
We don't need to generate another test_reg2, we could use the same from above. Thereby, we will not modify the existing tests and only have the new test instructions added in asmtest.out.h.